console/bridge.go

```diff
@@ -431,7 +431,11 @@ func (b *bridge) Send(call jsre.Call) (goja.Value, error) {
 				}
 			}
 		case rpc.Error:
-			setError(resp, err.ErrorCode(), err.Error())
+			errMap := map[string]interface{}{"code": err.ErrorCode(), "message": err.Error()}
+			if dataErr, ok := err.(rpc.DataError); ok {
+				errMap["data"] = dataErr.ErrorData()
+			}
+			resp.Set("error", errMap)
```

Comment:
We can cleanup by 

```go
+                       setError(resp, err.ErrorCode(), err.Error(), nil)
+               case rpc.DataError:
+                       setError(resp, err.ErrorCode(), err.Error(), err.ErrorData())
                default:
-                       setError(resp, -32603, err.Error())
+                       setError(resp, -32603, err.Error(), nil)
                }
                resps = append(resps, resp)
        }
@@ -457,8 +455,14 @@ func (b *bridge) Send(call jsre.Call) (goja.Value, error) {
        return result, nil
 }
 
-func setError(resp *goja.Object, code int, msg string) {
-       resp.Set("error", map[string]interface{}{"code": code, "message": msg})
+func setError(resp *goja.Object, code int, msg string, data interface{}) {
+       err := make(map[string]interface{})
+       err["code"] = code
+       err["message"] = msg
+       if data != nil {
+               err["data"] = data
+       }
+       resp.Set("error", err)
 }

```

Reply:
Agreed! (had to change the logic a bit, since rpc.Errors can also be rpc.DataErrors)